doc/config_migration_guide.md

```diff
@@ -0,0 +1,169 @@
+```
+-Dconfig.file=/path/to/config.conf
+```
+
+### Property migrations
+---
+#### Notes
+* No default values have been changed, so if you weren't specifying a property before you can continue to leave it unspecified and the value will not change.
+* The new configuration library supports different formats and many different features in configuration (substitutions, includes, overrides via environment variables, etc.).  The text here describes the 'plainest' way to define a new config file, but other methods are possible.  Read about the Typesafe config to learn more.
+* The example below assume that you will define a new configuration file called `application.conf`
+
+---
+##### Configuration file
+Old method: path to a JSON file via command line argument `--config=/path/to/config.json`
+New method: path to a HOCON .conf file passed via JVM arg: `-Dconfig.file=/path/to/config.conf`
+
+---
+##### Internal and external HTTP API ports
+Old method: command line arguments `--http-api-port` and `--internal-http-port`
+New method: in `application.conf`:
+```hocon
+jibri {
+    http {
+        external-api-port = <number>
+        internal-api-port = <number>
+    }
+}
+```
+---
+
+##### Single use mode
+Old method: in `config.json`:
+```json
+{
+    "single_use_mode": <Boolean>
+}
+```
+New method: in `application.conf`:
+```hocon
+jibri {
+    single-use-mode = <boolean>
+}
+```
+
+---
+##### Recordings directory
+Old method: in `config.json`:
+```json
+{
+    "recording_directory": <path>
+}
+```
+New method: in `application.conf`:
+```hocon
+jibri {
+    recording {
+        recordings-directory = <path>
+    }
+}
+```
+---
+##### Finalize script path
+Old method: in `config.json`:
+```json
+{
+    "finalize_recording_script_path": <path>
+}
+```
+New method: in `application.conf`:
+```hocon
+jibri {
+    recording {
+        finalize-script = <path>
+    }
+}
+```
+---
+##### Enabling statsD
+Old method: in `config.json`:
+```json
+{
+    "enable_stats_d": <boolean>
+}
+```
+New method: in `application.conf`:
+```hocon
+jibri {
+    stats {
+        enable-stats-d = <boolean>
+    }
+}
+```
+---
+##### XMPP environment config
+Old method: in `config.json`:
+```json
+{
+    "xmpp_environments": [
+        {
+            "name": <String>,
+            "xmpp_server_hosts": List<String>,
+            "xmpp_domain": <String>,
+            "control_login": {
+                "domain": <String>,
+                "username": <String>,
+                "password": <String>
+            },
+            "control_muc": {
+                "domain": <String>,
+                "room_name": <String>,
+                "nickname": <String>
+            },
+            "call_login": {
+                "domain": <String>,
+                "username": <String>,
+                "password": <String>
+            },
+            "room_jid_domain_string_to_strip_from_start": <String>,
+            "usage_timeout": <number>
+        },
+        { <more> }
+    }
```

Comment:
Missing `]}` (I think). It confuses the github interface, don't know if it affects te md file itself

Reply:
Fixed